### ISSUE TITLE
key (labelSet) already existed in dictionary

### ISSUE THREAD

**Issue by MaSUB** [SteeltoeOSS/Steeltoe#809]
## Describe the bug
Periodically we get the following exception:

```
System.ArgumentException: The key already existed in the dictionary.
   at System.Collections.Concurrent.ConcurrentDictionary`2.System.Collections.Generic.IDictionary<TKey,TValue>.Add(TKey key, TValue value)
   at Steeltoe.Management.OpenTelemetry.Metrics.MeasureMetricSdk`1.Bind(LabelSet labelset)
   at Steeltoe.Management.OpenTelemetry.Metrics.MeasureMetricSdk`1.Bind(IEnumerable`1 labels)
   at Steeltoe.Management.OpenTelemetry.Metrics.MeasureMetric`1.Record(SpanContext& context, T value, IEnumerable`1 labels)
   at Steeltoe.Management.Endpoint.Metrics.Observer.AspNetCoreHostingObserver.HandleStopEvent(Activity current, HttpContext arg)
   at Steeltoe.Management.Endpoint.Metrics.Observer.AspNetCoreHostingObserver.ProcessEvent(String evnt, Object arg)
   at Steeltoe.Common.Diagnostics.DiagnosticObserver.OnNext(KeyValuePair`2 event)
```

This seems to be tied to code in the class `CounterMetricSdkBase<T>` `Bind(LabelSet labelset, bool isShortLived)` when adding to the `_counterBoundInstruments` I believe this should be an `AddOrUpdate`. 

## Steps to reproduce 
Generally have just been allowing my app to run locally and it will eventually throw this exception without any outside input or requests to the API/service.

## Expected behavior
`labelSet` should be updated if already existing

## Environment (please complete the following information):
 - Steeltoe Version 3.1.0
 - Platform: Cloud Foundry 
 - OS: Linux
 - .NET Version .Net Core 3.1
 - Any other library versions to note

## Additional context or links
[Link to code in question](https://github.com/SteeltoeOSS/Steeltoe/blob/b443000ba28c78216b8e0b99aacfce26b64f3cb4/src/Management/src/OpenTelemetryBase/OpenTelemetry/Metrics/CounterMetricSdkBase.cs#L57)

**Comment by hananiel** [comment]
will be addressed in #299

**Comment by hananiel** [comment]
Resolved by https://github.com/SteeltoeOSS/Steeltoe/pull/828
